### PR TITLE
Add support for HTML `data-` attributes

### PR DESCRIPTION
**Pull request by gasi**
Usage:

```purescript
import Foreign.Object as FO

H.div
  { _data: FO.fromHomogenous { toggle: "buttons } }
  [...]
```

See individual commits for change descriptions.

Inspired by `purescript-react-basic`.

See: https://github.com/collegevine/purescript-elmish/pull/12

### CODE REVIEW

File: codegen/index.js [hunk 1]
```diff
@@ -76,7 +79,8 @@ const propType = (e, p) => {
 const printRow = (e, elProps) =>
   elProps.length > 0
     ? `
-  ( ${elProps.map(p => `${p} :: ${propType(e, p)}`).join("\n  , ")}
+  ( _data :: Object String
```
Comment (fsoikin) [inline codegen/index.js:82]: `Object String` will not work with the intended usage of `_data: { foo: "bar" }`
To convert the record to `Object String`, you need `fromHomogeneous`.

**Review by fsoikin — CHANGES_REQUESTED** [review]
Overall looks fine, except the usage example probably needs `fromHomogenous` (or am I missing something?), plus remove the Spago stuff

File: codegen/index.js (hunk 1 above)
Comment (gasi) [reply codegen/index.js:82]: Yep, I fixed that 👍

**Comment by gasi** [comment]
@fsoikin Thanks for your review. I addressed the feedback and this is ready for final review.

**Review by fsoikin — APPROVED** [review]
Looks good!

**Pull request merged**